// ng-zorro-antd/src/core/overlay/scroll/nz-block-scroll-strategy.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var NzBlockScrollStrategy = /** @class */ (function () {
    function NzBlockScrollStrategy(document, renderer) {
        this.document = document;
        this.renderer = renderer;
    }
    /**
     * @return {?}
     */
    NzBlockScrollStrategy.prototype.attach = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    NzBlockScrollStrategy.prototype.enable = /**
     * @return {?}
     */
    function () {
        this.renderer.setStyle(document.body, 'overflow', 'hidden');
    };
    /**
     * @return {?}
     */
    NzBlockScrollStrategy.prototype.disable = /**
     * @return {?}
     */
    function () {
        this.renderer.removeStyle(document.body, 'overflow');
    };
    return NzBlockScrollStrategy;
}());
export { NzBlockScrollStrategy };
if (false) {
    /** @type {?} */
    NzBlockScrollStrategy.prototype.document;
    /** @type {?} */
    NzBlockScrollStrategy.prototype.renderer;
}
